xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/ListenerProvider.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.listener;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.rendering.listener.chaining.ChainingListener;
+import org.xwiki.rendering.listener.chaining.ListenerChain;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Allows to access a listener instance.
+ *
+ * @version $Id$
+ * @since 15.3RC1
+ * @since 14.10.8
+ */
+@Role
+@Unstable
+public interface ListenerProvider
+{
+    /**
+     * @param action the action performed by the listener ("render" or "parse")
+     * @param syntaxHint the hint of the syntax using for the action
+     * @return {@code true} when the listener provider can return a listener for the given action and syntaxHint
+     */
+    boolean accept(String action, String syntaxHint);
```

Comment:
Wouldn't it be better to either use an enum for the action or at least provide constants for the two possible values directly in `ListenerProvider`?

Reply:
The constants are here https://github.com/xwiki/xwiki-rendering/pull/258/files#diff-a69d93e25fa580489bfec51a5d51a405b7ad87f7f2f5b8ad0592f768a2aea9d2R49-R54
I considered the enum but I'm not sure of the pros and cons here.

Reply:
Yes, I've seen that you've provided the constants in `ListenerRegistry`, but `ListenerRegistry` is internal so any extension that wants to provide a `ListenerProvider` either needs to depend on this internal class or re-define the constants. Now I'm wondering if `ListenerRegistry` shouldn't also be public API as it might be interesting to use for any extension that implements a parser or renderer.

Reply:
Indeed, I meant to have this constants public so either:
- moved them to `ListenerProvider`
- make `ListenerRegistry` public

Reply:
Constants moved to `ListenerProvider`.